runtime/meta.go

```diff
@@ -0,0 +1,138 @@
+package encore
+
+import (
+	"net/url"
+	"time"
+
+	"encore.dev/runtime/config"
+)
+
+var apiBaseURL *url.URL
+
+// Meta returns metadata about the running application.
+//
+// Meta will never return nil.
+func Meta() *AppMetadata {
+
+	return &AppMetadata{
+		AppID:      config.Cfg.Runtime.AppSlug,
+		APIBaseURL: *apiBaseURL,
+		Environment: EnvironmentMeta{
+			Name:  config.Cfg.Runtime.EnvName,
+			Type:  EnvironmentType(config.Cfg.Runtime.EnvType),
+			Cloud: CloudProvider(config.Cfg.Runtime.EnvCloud),
+		},
+		Build: BuildMeta{
+			Revision:           config.Cfg.Static.AppCommit.Revision,
+			UncommittedChanges: config.Cfg.Static.AppCommit.Uncommitted,
+		},
+		Deploy: DeployMeta{
+			ID:   config.Cfg.Runtime.DeployID,
+			Time: config.Cfg.Runtime.DeployedAt,
+		},
+	}
+}
+
+func init() {
+	if config.Cfg != nil {
```

Comment:
I assume this check is only to work outside of Encore apps? Maybe worth documenting